arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/memoization.kt

```diff
@@ -124,13 +126,13 @@ private class MemoizedHandler<F, in K : MemoizedCall<F, R>, out R>(val f: F) {
         // No value is set yet (race condition)
         if (bb == null) {
           // Value was set, return the value
-          if (cache.compareAndSet(old, old + Pair(k, b))) return b
+          if (cache.compareAndSet(old, old + Pair(k, Option.fromNullable(b)))) return b
```

Comment:
```suggestion
          if (cache.compareAndSet(old, old + Pair(k, Some(b)))) return b
```
Shouldn't this be `Some(b)`, and instead of using `Map.get` in line 125 and 120 it should use `getOrNone`.
That way it can pattern match of `Option`:
 - `None` meaning the value wasn't present for the key `K`.
 - `Some` if a value was found for `k`, and return it.